base/strings/search.jl

```diff
@@ -149,7 +153,7 @@ _nthbyte(a::Union{AbstractVector{UInt8},AbstractVector{Int8}}, i) = a[i]
 
 function _searchindex(s::String, t::String, i::Integer)
     # Check for fast case of a single byte
-    lastindex(t) == 1 && return something(findnext(isequal(t[1]), s, i), 0)
+    sizeof(t) == 1 && return something(findnext(isequal(first(t)), s, i), 0)
```

Comment:
These should be equivalent, right?

Reply:
I don't think it is quite:
* `sizeof(t) == 1` tests that `t` has one code unit. It should probably be written as `ncodeunits(t) == 1`, which just happens to be the same for `String`.
* `lastindex(t) == 1` tests that `t` has only one character since the last valid index is at the start of the string. It could equivalently be written as `length(t) == 1` which might be a bit more efficient.

It might make sense to special-case `ncodeunits(t) == 1` since that implies `length(t) == 1` and is more efficient to compute, so `(ncodeunits(t) == 1 || length(t) == 1)` could be more efficient? Or not since it's more complicated.

Reply:
Hm, but why is the comment `# Check for fast case of a single byte` then?